packages/rest/src/http/route.ts

```diff
@@ -573,7 +573,8 @@ function validateRouteUnique(diagnostics: DiagnosticCollector, operations: Opera
 const resourceOperationToVerb: any = {
   read: "get",
   create: "post",
-  createOrUpdate: "put",
+  createOrUpdate: "patch",
```

Comment:
is `createOrUpdate` the right name for the patch one then?

Reply:
Per [RFC7231](https://httpwg.org/specs/rfc7231.html#PUT), PUT is described as creating or replacing a target resource:

> The PUT method requests that the state of the [target resource](https://httpwg.org/specs/rfc7231.html#resources) be created or replaced with the state defined by the representation enclosed in the request message payload.

Internally, we've used either `createOrUpdate` or `upsert` for this semantic in our data plane client libraries. The only downside with using `patch` for createOrUpdate (from an Azure/client library/our team's perspective) is that historically, we've used the createOrUpdate for ARM libraries, and ARM uses `put`, but implemented it more like `patch`. So there is history to fight here, but I don't think that we should bleed that out to Cadl-lang libraries...